### PR TITLE
feat: pause/unpause game

### LINKED ISSUE #26: feat: pause/unpause game

### PR DESCRIPTION
**Pull request by joaomagfreitas**

### CODE REVIEW

File: src/models/domain/game.gd [hunk 1]
```diff
@@ -20,3 +21,23 @@ func _init(
...
+func end():
+	self.status = GameStatus.finshed
+	
+func is_paused():
+	return self.status == GameStatus.paused
+
+enum GameStatus {
+	resumed,
+	paused,
+	finshed
```
Comment (rutesantos4) [inline src/models/domain/game.gd]:
```suggestion
	finished
```

File: src/nodes/pause_button.gd [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+var game: Game
+
+func _ready():
+	game_node = get_tree().current_scene
+	
+	pausable_group = game_node.get_node("Pausable")
+	pause_menu = game_node.get_node("Stickable/PauseMenu")
+	game = SceneSwitcher.get_game_state()
+	
+	self.connect("pressed", _on_game_paused)
```
Comment (rutesantos4) [inline src/nodes/pause_button.gd]: I think you could have used the signals presented in the node section, but this is fine
image

File: src/nodes/pause_menu.gd [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	game = SceneSwitcher.get_game_state()
+	game_node = get_tree().current_scene
+	
+	resume_button_node = $MarginContainer/VBoxContainer/Resume
+	mute_button_node = $MarginContainer/VBoxContainer/Mute
+	quit_button_node = $MarginContainer/VBoxContainer/Quit
+	
+	resume_button_node.connect("pressed", resume)
+	mute_button_node.connect("pressed", mute)
+	quit_button_node.connect("pressed", quit)
```
Comment (rutesantos4) [inline src/nodes/pause_menu.gd]: same here

File: src/models/domain/game.gd (hunk 1 above)
Comment (joaomagfreitas) [reply src/models/domain/game.gd]: nice catch!

File: src/nodes/pause_button.gd (hunk 2 above)
Comment (joaomagfreitas) [reply src/nodes/pause_button.gd]: I completely forgot about that, thank you!

**Review by rutesantos4 — APPROVED** [review]

**Pull request merged**